### DISCUSSION TITLE
Async pipe is not pure 🤯 - justangular.com

### DISCUSSION THREAD

**Discussion by eneajaho** [eneajaho/just-angular#6] [General]
https://justangular.com/blog/async-pipe-is-not-pure

**Comment by stickyamp** [comment]
How does this implementation work if the new custom pipe is returning this.latestValue in sync way while the assignation is made in async via .subscribe ?

  this.sub = obs.subscribe({
    next: (res) => {
      this.latestValue = res;
      this.cdr.markForCheck();
    },

Does it has to do with the angular change detection that auto fixes it or something?

**Reply by eneajaho** [reply]
> How does this implementation work if the new custom pipe is returning this.latestValue in sync way while the assignation is made in async via .subscribe ?
> 
>   this.sub = obs.subscribe({
>     next: (res) => {
>       this.latestValue = res;
>       this.cdr.markForCheck();
>     },
> 
> Does it has to do with the angular change detection that auto fixes it or something?

Yes

**Reply by stickyamp** [reply]
Thanks for replying! 

Could you add a little more detail so i get full knowledge about this behavior? 

Thank you

**Reply by eneajaho** [reply]
> Thanks for replying! 
> 
> Could you add a little more detail so i get full knowledge about this behavior? 
> 
> Thank you

Sure. I recommend reading https://justangular.com/blog/a-change-detection-zone-js-zoneless-local-change-detection-and-signals-story as it includes everything you need to understand how change detection and async pipe works too.
